homeassistant/components/switcher_kis/entity.py

```diff
@@ -18,3 +26,57 @@ def __init__(self, coordinator: SwitcherDataUpdateCoordinator) -> None:
         self._attr_device_info = DeviceInfo(
             connections={(dr.CONNECTION_NETWORK_MAC, coordinator.mac_address)}
         )
+
+    async def _async_call_api(self, api: str, *args: Any) -> None:
+        """Call Switcher API."""
+        _LOGGER.debug("Calling api for %s, api: '%s', args: %s", self.name, api, args)
+        response: SwitcherBaseResponse | None = None
+        error = None
+
+        try:
+            async with SwitcherApi(
+                self.coordinator.data.device_type,
+                self.coordinator.data.ip_address,
+                self.coordinator.data.device_id,
+                self.coordinator.data.device_key,
+                self.coordinator.token,
+            ) as swapi:
+                response = await getattr(swapi, api)(*args)
+        except (TimeoutError, OSError, RuntimeError) as err:
+            error = repr(err)
+
+        if error or not response or not response.successful:
+            _LOGGER.error(
+                "Call api for %s failed, api: '%s', args: %s, response/error: %s",
+                self.coordinator.name,
+                api,
+                args,
+                response or error,
+            )
+            self.coordinator.last_update_success = False
```

Comment:
This method is legacy and does not raise error in case of failure and should not be used, please remove it
```suggestion
```